akka-actor/src/main/scala/akka/io/Dns.scala

```diff
@@ -89,3 +89,11 @@ class DnsExt(system: ExtendedActorSystem) extends IO.Extension {
 
   def getResolver: ActorRef = manager
 }
+
+object IpVersionSelector {
+  def getInetAddress(ipv4: Option[Inet4Address], ipv6: Option[Inet6Address]): Option[InetAddress] =
+    sys.props.get("java.net.preferIPv6Addresses") match {
```

Comment:
I used the system property directly because, in theory, the properties can change dynamically in runtime. This is not our use case and seems also very concrete but that is the reason. A `val` will require an application restart to take the changes.

> Question is if this really needs a tests. Pretty trivial logic here.

Not sure if you are talking about the test cases, if it can be tested I think that it should be included.